app/src/main/java/de/westnordost/streetcomplete/quests/drinking_water/AddDrinkingWater.kt

```diff
@@ -0,0 +1,36 @@
+package de.westnordost.streetcomplete.quests.drinking_water
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.osmquest.OsmFilterQuestType
+
+class AddDrinkingWater : OsmFilterQuestType<DrinkingWater>() {
+
+    override val elementFilter = """
+        nodes with (
+            man_made = water_tap
+            or man_made = water_well
+            or amenity = toilets
+            or natural = spring
+        )
+        and access !~ private|no and indoor != yes
+        and !drinking_water and amenity != drinking_water
```

Comment:
Maybe even exclude any nodes with `drinking_water:legal`?

```suggestion
        and !drinking_water and !drinking_water:legal and amenity != drinking_water
```

Reply:
yes. If only `drinking_water:legal` is set but not `drinking_water`, it will have a good reason